core/shared/src/main/java/de/mm20/launcher2/search/location/OpeningSchedule.kt

```diff
@@ -0,0 +1,26 @@
+package de.mm20.launcher2.search.location
+
+import de.mm20.launcher2.serialization.DurationSerializer
+import de.mm20.launcher2.serialization.LocalTimeSerializer
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.serialization.Serializable
+import java.time.DayOfWeek
+import java.time.Duration
+import java.time.LocalTime
+
+@Serializable
+data class OpeningHours(
+    val dayOfWeek: DayOfWeek,
+    @Serializable(with = LocalTimeSerializer::class)
+    val startTime: LocalTime,
+    @Serializable(with = DurationSerializer::class)
+    val duration: Duration
+) {
+    override fun toString(): String = "$dayOfWeek $startTime-${startTime.plus(duration)}"
+}
+
+@Serializable
+data class OpeningSchedule(
+    val isTwentyFourSeven: Boolean,
+    val openingHours: ImmutableList<OpeningHours>
+)
```

Comment:
maybe 
```kotlin
sealed class OpeningSchedule {
    data object TwentyFourSeven: OpeningSchedule()
    data class Hours(val openingHours: ImmutableList<OpeningHours>): OpeningSchedule()
}
```
as they should be mutually exclusive?

Reply:
Yes, good point (but I would use a `sealed interface`). Though that would probably break backwards compatibility with already stored locations, but I think that's okay since the item is updated anyways as soon as the user views it.
Also I have replaced `ImmutableList` with a standard `List` because it doesn't support serialization.